### PR TITLE
spec: clarify expectations for Identity RPCs

### PR DESCRIPTION
**Pull request by jdef**
related to #65

### CODE REVIEW

File: spec.md [hunk 1]
```diff
@@ -289,17 +289,21 @@ The error code `OPERATION_PENDING_FOR_VOLUME` may be returned by the plugin in t
 
 #### `GetSupportedVersions`
 
-A Plugin SHALL reply with a list of supported CSI versions. The initial version of the CSI specification is 0.1.0 (in *major.minor.patch* format).
+A Plugin SHALL reply with a list of supported CSI versions.
+The initial version of the CSI specification is 0.1.0 (in *major.minor.patch* format).
 A CO MAY execute plugin RPCs in the manner prescribed by any such supported CSI version.
 The versions returned by this call are orthogonal to any vendor-specific version metadata (see `vendor_version` in `GetPluginInfoResponse`).
 
+NOTE: Changes to this RPC should be approached very conservatively since the request/response protobufs here are critical for proper client-server version negotiation.
```
Comment (akutz) [inline spec.md]: Hi @jdef,

It's looking good! I wonder though, this RPC is so minimal I'd just as soon as declare it immutable and any possible enhancements be added as a new RPC, ex. `GetSupportedVersionsExRequest`.

File: spec.md [hunk 2]
```diff
@@ -311,7 +315,7 @@ message GetSupportedVersionsResponse {
   }
 }
 
-// Specifies the version in Semantic Version 2.0 format.
+// Specifies a version in Semantic Version 2.0 format.
```
Comment (akutz) [inline spec.md]: Hi @jdef,

Perhaps link to the SemVer site?

`// Specifies a version in Semantic Version 2.0 (http://semver.org) format.`

File: spec.md (hunk 1 above)
Comment (jdef) [inline spec.md]: Thanks for the feedback. I'm not crazy about immutable protos. We purposefully didn't declare fields using `required` syntax, and instead included `REQUIRED` and `OPTIONAL` in the docs for each field. I'm also not crazy about the `XyzExRequest` nomenclature.

File: spec.md (hunk 2 above)
Comment (jdef) [inline spec.md]: Thanks, added.

**Pull request merged**